tools/tpm2_createpolicy.c

```diff
@@ -0,0 +1,333 @@
+            LOG_INFO("PCR Index= %d\n",ctx->pcr_index);
+            UINT32 rval = build_policy(ctx, build_pcr_policy);
+            if (rval != TPM_RC_SUCCESS) {
+                LOG_ERR("Failed build_policy\n");
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+static bool init(int argc, char *argv[], char *envp[], createpolicypcr_ctx *ctx) {
+    
+    struct option sOpts[] = {
+            { "policy-file",     required_argument,  NULL,  'f' },
+            { "policy-PCR",      no_argument,        NULL,  'P' },
+            { "pcr-index",       required_argument,  NULL,  'i' },
+            { "pcr-alg",         required_argument,  NULL,  'g' },
+            { "pcr-input-file",  required_argument,  NULL,  'F' },
+            { NULL,              no_argument,        NULL, '\0' },
+    };
+
+    if (argc == 1) {
+        execute_man(argv[0], envp);
+        return false;
+    }
+
+    if (argc > (int) (2 * sizeof(sOpts) / sizeof(struct option))) {
+        showArgMismatch(argv[0]);
+        return false;
+    }
+
+    int opt;
+    bool result;
+    while ((opt = getopt_long(argc, argv, "Pf:i:g:F:", sOpts, NULL)) != -1) {
+        switch (opt) {
+        case 'f':
+            ctx->policy_file_flag=true;
+            snprintf(ctx->policyfile, sizeof(ctx->policyfile), "%s", optarg);
+            break;
+        case 'P':
+            ctx->policy_type_pcr_flag = true;
+            printf("Policy type chosen is policyPCR.\n");
+            break;       
+        case 'i':
+            ctx->pcr_index_flag = true;
+            if(string_bytes_get_uint32(optarg, &ctx->pcr_index)!=true) {
+                return false;
+            }
+            if(ctx->pcr_index > 23 || ctx->pcr_index<0) {
+                printf("Invalid pcr_index %d. Choose between 0..23\n",
+                    ctx->pcr_index );
+                return false;
+            }
+            break;
+        case 'F':
+            ctx->raw_pcr_flag = true;
+            snprintf(ctx->raw_pcr_file, sizeof(ctx->raw_pcr_file), "%s", optarg);
+            break;
+        case 'g':
+            if(string_bytes_get_uint16(optarg, &ctx->hash_alg)!=true) {
+                return false;
+            }
+            break;
+        case '?':
+        default:
+            showArgMismatch(argv[0]);
+            return false;
+        }
+    }
+    
+    return true;
+}
+
+int execute_tool(int argc, char *argv[], char *envp[], common_opts_t *opts,
+        TSS2_SYS_CONTEXT *sapi_context) {
+
+    /* opts is unused */
+    (void) opts;
+
+    createpolicypcr_ctx ctx = {
+            .sapi_context = sapi_context,
+            .policy_session = malloc(sizeof(SESSION)),
+            .policy_digest = { 0 },
+            .pcr_index = 0,
+            .hash_alg = TPM_ALG_SHA1,
+            .policy_type_pcr_flag = false,
+            .policy_file_flag = false,
+            .pcr_index_flag = false,
+            .raw_pcr_flag = false
```

Comment:
A lot of flags... group them, makes it read better
struct {
  bool pcr;
  bool file;
  ...
} flags;

ctx->flags.policy 

You can even do bitfields and save us a little space:

struct {
  UINT8 raw : 1;
  UINT8 policy : 1;
  ...
} flags;